core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -3220,86 +3169,136 @@ private boolean syncMetadata(JournalContext journalContext, LockedInodePath inod
     return true;
   }
 
-  private boolean syncChildrenMetadata(JournalContext journalContext, LockedInodePath inodePath,
-      DescendantType syncDescendantType, BlockDeletionContext blockDeletionContext)
-      throws FileDoesNotExistException, InvalidPathException, IOException,
-      DirectoryNotEmptyException {
-    if (syncDescendantType == DescendantType.NONE) {
-      return false;
+  /**
+   * This class represents the result for a sync.
+   */
+  private static class SyncResult {
+    private boolean mLoadMetadata;
```

Comment:
it's tricky how `syncInodeMetadata` performs delete sync but doesn't do metadata load sync, instead returning a boolean saying whether the caller needs to load metadata afterwards. Can we comment on this in the `SyncResult` type and in the doc for `syncInodeMetadata`?

Reply:
Yeah, we have a multi-step process for syncing, primarily because of how deleting an inode requires a write lock (but loading metadata does not). I added comments.